### PR TITLE
Adds Gotify (new notifier)

### PR DESCRIPTION
**Pull request by DmitrySandalov**
Proposed changes in this pull request:
- Adds new notifier, Gotify 
    - https://gotify.net
    - https://github.com/gotify/server

- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read [contribution guide](https://github.com/SickChill/SickChill/blob/master/.github/CONTRIBUTING.md)

## Summary by CodeRabbit

- **New Features**
  - Integrated Gotify notification service for real-time updates.
  - Added configuration options for Gotify in the notification settings.
  - Implemented a test function to verify Gotify notification setup.

- **Enhancements**
  - Introduced new UI elements for enabling and configuring Gotify notifications.
  - Added a dedicated Gotify icon for better visual identification.

- **Bug Fixes**
  - Ensured proper handling of Gotify notification settings within the application's configuration management.

### CODE REVIEW

File: sickchill/oldbeard/notifiers/gotify.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            r = requests.post(
+                f'{host}message?token={token or settings.GOTIFY_AUTHORIZATIONTOKEN}',
+                data=json.dumps(dict(title=title, message=f"{title} : {msg}")),
+                headers=headers,
+            )
+            r.raise_for_status()
+        except Exception as error:
+            logger.exception(f"Error Sending Gotify message: {error}")
+            return False, error
+        return True, None
```
[the remark this answers is not in the document]
Comment (DmitrySandalov) [reply sickchill/oldbeard/notifiers/gotify.py]: Fixed in commit [a2caa445e](https://github.com/SickChill/sickchill/pull/8651/commits/a2caa445e4bd451c3e27251db16017a630eead6a).

File: sickchill/gui/slick/js/core.js [hunk 2]
```diff
@@ -1079,6 +1079,39 @@ const SICKCHILL = {
...
+                authorizationToken: gotify.authToken,
+              }).done(data => {
+                $('#testGotify-result').html(data);
+                $('#testGotify').prop('disabled', false);
+              });
+            });
+
             $('#testPushbullet').on('click', function () {
                 const pushbullet = {};
                 pushbullet.api = $.trim($('#pushbullet_api').val());
```
[the remark this answers is not in the document]
Comment (DmitrySandalov) [reply sickchill/gui/slick/js/core.js:1117]: Aligns with other notifier's code.

File: sickchill/oldbeard/notifiers/gotify.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        :returns: the message to send
+        """
+
+        if not (force or settings.USE_GOTIFY):
+            logger.debug("Notification for Gotify not enabled, skipping this notification")
+            return False, "Disabled"
+
+        logger.debug("Sending a Gotify message {0}".format(message))
+
+        return self._send_gotify_msg(title, message, host, token)
```
[the remark this answers is not in the document]
Comment (DmitrySandalov) [reply sickchill/oldbeard/notifiers/gotify.py]: Aligns with other notifier's code.

Is there anything from this list we should fix?

**Comment by miigotu** [comment]
@DmitrySandalov can the token go inside the post data? I don't see why we would post and have get arguments, exposing the token in logs (especially if someone is using https for their gotify)

Also, see https://github.com/SickChill/sickchill/actions/runs/7230779189/job/19703381110?pr=8651 and change the single quotes to double quotes around the request url parameter

**Comment by DmitrySandalov** [comment]
@miigotu Seems like it's not possible to put the token to post data (ref their [Swagger](https://gotify.net/api-docs#/) and [sources](https://github.com/gotify/server/blob/master/auth/authentication.go)), but a header can be utilized alternatively.
> The token can be transmitted in a header named X-Gotify-Key, in a query parameter named token or
> through a header named Authorization with the value prefixed with Bearer (Ex. Bearer randomtoken).

I chose `Authorization: Bearer randomtoken` (commit [483aacfd6](https://github.com/SickChill/sickchill/pull/8651/commits/483aacfd66a753fd446b6048e592a571f1fb9a9e)). `black` error was fixed as well.

**Pull request merged**